internal/controllers/lvmcluster/controller.go

```diff
@@ -194,7 +192,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	result, reconcileError := r.reconcile(ctx, lvmCluster)
 
-	statusError := r.updateLVMClusterStatus(ctx, lvmCluster)
+	statusError := r.updateLVMClusterStatus(ctx, lvmCluster, reconcileError)
```

Comment:
Could you explain to me again Why its necessary to do an explicit Status error check here? I was expecting that part of reconcile would be to update the status. Now its still 2 different functions, and its hard to incorporate new status elements. Similarly the below check on the reconcileError is not synchronized with the statusError, we wanted to have this error be one I think?